### ISSUE TITLE
BigQuery CTE not recognised when quoting CTE name in FROM

### ISSUE THREAD

**Issue by stefandoorn** [sqlfluff/sqlfluff#6556]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

We do have a BigQuery using a CTE, but rule ST03 seems to report it as not used, while it is being used but it does seem to be related to the quoting of the CTE name. 

We are using `jinja2` for templates. The dialect is BigQuery.

I'm happy to help out on a PR, but have never worked in the SQLFluff codebase yet.

### Expected Behaviour

I expected a PASS on ST03 in the below observed behavior, even if the CTE name is quoted.

### Observed Behaviour

```bash
sqlfluff lint
==== sqlfluff ====
sqlfluff:                3.3.0 python:                 3.12.2
implementation:        cpython verbosity:                   1
dialect:              bigquery templater:               jinja
rules:                                all
==== readout ====

....
                                                                                                          
== [/usr/src/app/sql.sql.jinja2] FAIL                                                                                                                                                                 
L:   2 | P:   3 | ST03 | Query defines CTE "cte_name" but does not use it.                                                                                                                                                                                            
                       | [structure.unused_cte]

...
```


### How to reproduce

We have a pretty lengthy query but I brought it back to the example test-case here:

```sql
WITH
  cte_name AS (SELECT 1)

SELECT
  *
FROM `cte_name`
```

It does work though (or well, ST03 doesn't report an error) if it's written like this:

```sql
WITH
  cte_name AS (SELECT 1)

SELECT
  *
FROM cte_name
```

Both queries work fine within BigQuery.

### Dialect

BigQuery

### Version

3.3.0

### Configuration

```toml
[tool.sqlfluff.core]
verbose = 1
templater = "jinja"
sql_file_exts = ".sql,.sql.jinja2"
dialect = "bigquery"
exclude_rules = "LT02,LT05,LT12,RF01,LT01,CP02,RF04,CV02,ST07,RF02,CV03,LT09,ST06"

[tool.sqlfluff.templater.jinja.context]
environmentName = "test"
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by stefandoorn** [comment]
Thanks!
